<?xml version="1.0" encoding="UTF-8"?>
<launch>
  <arg name="pointcloudTopic" default="/velodyne_points"/>
  <arg name="pointcloudFrame" default="odom"/>

  <!-- <node pkg="test_octomap" type="tf_camera_to_map" name="tf_camera2map"/> -->

  <!-- <node pkg="test_octomap" type="pointcloud2map" name="pointcloud2map" output="screen">
    <param name="pointcloud_trans_topic" value="$(arg pointcloudTopic)">
    <param name="pointcloud_in_topic" value="/camera/depth/points">
  </node> -->

  <rosparam file="$(find octomap_scout)/config/octomap_server_params.yaml" command="load" />

  <!-- **octomap config** -->
  <node pkg="octomap_server" type="octomap_server_node" name="octomap_server">
    <!-- name of the fixed frame, needs to be "/map" for SLAM -->
    <param name="frame_id" type="string" value="$(arg pointcloudFrame)" />

    <!-- topic from where pointcloud2 messages are subscribed -->
    <remap from="/cloud_in" to="$(arg pointcloudTopic)" />
  </node>

  <!-- **costmap config** -->
  <!--node pkg="test_octomap" type="turtlebot_navigation" name="turtlebot_navigation_node" output="screen">
    <rosparam file="$(find octomap_scout)/config/costmap_common_params.yaml" command="load" ns="global_costmap" />
    <rosparam file="$(find octomap_scout)/config/costmap_common_params.yaml" command="load" ns="local_costmap" />
    <rosparam file="$(find octomap_scout)/config/local_costmap_params.yaml" command="load" />
    <rosparam file="$(find octomap_scout)/config/global_costmap_params.yaml" command="load" />
    <param name="global_costmap/global_frame" value="/odom"/>
    <param name="global_costmap/robot_base_frame" value="/base_link"/>
    <param name="global_costmap/point_cloud_sensor/sensor_frame" value="/camera_depth_frame"/>
    <param name="global_costmap/point_cloud_sensor/topic" value="/camera/depth/points"/>    
    <param name="local_costmap/global_frame" value="/odom"/>
    <param name="local_costmap/robot_base_frame" value="/base_link"/>
    <param name="local_costmap/point_cloud_sensor/sensor_frame" value="/camera_depth_frame"/>
    <param name="local_costmap/point_cloud_sensor/topic" value="/camera/depth/points"/>
    <param name="local_costmap/obstacle_layer/point_cloud_sensor/topic" value="/camera/depth/points"/>
    <remap from="map" to="/projected_map"/>
  </node-->

   <!-- RViz -->
  <node pkg="rviz" type="rviz" name="$(anon rviz)" respawn="false" output="screen" args="-d $(find octomap_scout)/rviz/scout_octomap.rviz"/>
</launch>
